### ISSUE TITLE
No longer ES5 compatible due to use of Symbol

### ISSUE THREAD

**Issue by Schnark** [josdejong/mathjs#2245]
The documentation says that math.js is still ES5 compatible. But trying to execute it in an old browser throws errors, because `Symbol` is undefined (which is ES6). Actually, `Symbol` is used in many places in the library, but in most cases there is a check to see whether it exists before it is accessed, but in three places this check is missing. As far as I can see (such old browsers aren't good for debugging, I am only told all 3 instances are in line 40 of https://cdnjs.cloudflare.com/ajax/libs/mathjs/9.4.2/math.js), these are the 3 instances: https://github.com/josdejong/mathjs/search?q=Symbol.iterator (Matrix, SparseMatrix, DenseMatrix)

Either you should guard them with a check for `typeof Symbol !== 'undefined'`, or update the documentation to say that an ES6 compliant browser is required.

**Comment by Schnark** [comment]
I tested this with an old Firefox, I assume something around version 20 (on an offline computer I sometimes use, so I currently can't tell you the exact version).

The code at the top of the file doesn't seem to be a polyfill for `Symbol`, but merely internal uses of `Symbol` _provided it exists_:

```js
"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"})
```
just means: If `Symbol` is defined, and `Symbol.toStringTag` exists, then use it to output `[object Module]` when `e` is converted to a string.

**Comment by josdejong** [comment]
I've tested on IE 11, it indeed doesn't work with Symbol related errors:

![afbeelding](https://user-images.githubusercontent.com/568626/121769698-73363580-cb65-11eb-9d8b-b6017148033c.png)

So we should look into this, see whether it's fixable with babel (Or decide to drop support for IE if it's becoming to big of a headache ([a bit too early maybe](https://death-to-ie11.com/))

**Comment by josdejong** [comment]
It could be that the karma tests running on IE 11 are transpiled/bundled in a different way than the bundle `math.min.js` - it's odd that the tests succeed but the bundle doesn't work.

**Comment by josdejong** [comment]
@m93a are you able to look into this issue maybe?

**Comment by m93a** [comment]
I'll see what I can do 👍

**Comment by m93a** [comment]
I've created PR #2266 which should solve this problem, however I don't have any way to run IE11 to test it. @josdejong @Schnark Could you please build the PR, test it in IE11 and let me know if it worked?

**Comment by josdejong** [comment]
Fix published now in `v9.4.4`
